src/sounding.F90

```diff
@@ -320,29 +327,107 @@ subroutine read_sounding( gr, iunit, runtype, p_sfc, zm_init,&
         end if  ! k > nlevels
 
         ! Regular situation w/ linear int.
-        IF ( trim( runtype ) /= "dycoms2_rf02" ) THEN
-
-          um(i)   = lin_interpolate_two_points( gr%zt(1,i), z(k), z(k-1), u(k), u(k-1) )
-          vm(i)   = lin_interpolate_two_points( gr%zt(1,i), z(k), z(k-1), v(k), v(k-1) )
-          ugm(i)  = lin_interpolate_two_points( gr%zt(1,i), z(k), z(k-1), ug(k), ug(k-1) )
-          vgm(i)  = lin_interpolate_two_points( gr%zt(1,i), z(k), z(k-1), vg(k), vg(k-1) )
-          thlm(i) = lin_interpolate_two_points( gr%zt(1,i), z(k), z(k-1), theta(k), theta(k-1) )
-          rtm(i)  = lin_interpolate_two_points( gr%zt(1,i), z(k), z(k-1), rt(k), rt(k-1) )
-          press(i) = lin_interpolate_two_points( gr%zt(1,i), z(k), z(k-1), p_in_Pa(k), p_in_Pa(k-1) )
-          wm(i) = lin_interpolate_two_points( gr%zt(1,i), z(k), z(k-1), subs(k), subs(k-1) )
-
-          if ( sclr_dim > 0 ) then
-            do j = 1, sclr_dim
-              sclrm(i,j) = lin_interpolate_two_points( gr%zt(1,i), z(k), z(k-1),  & 
-                                    sclr(k,j), sclr(k-1,j) )
-            end do
-          end if
-          if ( edsclr_dim > 0 ) then
-            do j = 1, edsclr_dim
-              edsclrm(i,j) = lin_interpolate_two_points( gr%zt(1,i), z(k), z(k-1),  & 
-                                      edsclr(k,j), edsclr(k-1,j) )
-            end do
-          end if
+        IF ( trim( runtype ) /= "dycoms2_rf02" .or. l_modify_ic_for_cnvg_test ) THEN
```

Comment:
Do we only need to smooth the sounding for the dycoms2_rf02 case?  Do the wangara, rico, and bomex cases converge without smoothing the initial condition?

Reply:
@vlarson: for our convergence test simulation in the paper, we need to smooth the initial condition profiles for all the cases:
1. For RICO, BOMEX and Wangara case, we employ cubic spline interpolation to remap the sounding profiles to the CLUBB's grid to obtain smoother initial condition profiles 
2. For Dycoms_RF02 case: the default setup in CLUBB uses the formulations built in the src/sounding.F90 to derive the initial conditions, which are space-dependent. In our investigation, I choose to construct the initial condition profile using the same formulations in the code, and saved them in  **dycoms2_rf02_sounding.in** file. Then all simulations read the **dycoms2_rf02_sounding.in**  file for initialization  
3. there are also some changes for *_soundign.in files, I choose to do the changes in the simulation run scripts.  I will submit a separate pull request for the scripts to run convergence test simulations